server/src/entities/asset-face.entity.ts

```diff
@@ -37,6 +41,9 @@ export class AssetFaceEntity {
   @Column({ default: 0, type: 'int' })
   boundingBoxY2!: number;
 
+  @Column({ nullable: true, type: 'varchar' })
+  sourceType?: string | null;
```

Comment:
Another option is to make it non-nullable and default it to another enum value like MACHINE_LEARNING

Reply:
I prefer not to since it's hundreds of thousands of strings that are useless in the vast majority of Immich instances. But `ml` (rather than `machine-learning`) would at least minimize that.

Reply:
If you care about size use a numeric enum value. The enum label in the code can be machine learning still.

Reply:
I'm fine with that.

Reply:
Is it possible to have the source type be an int in the database, but use enum in the rest of the codebase, including the open-api-spec and subsequently generated client code?

Reply:
Hi @jrasm91, I think it can be achieved by using `enum` open-api-spec and defining a datatype at the database level? If so, the DB can be queries using instead of fixed numbers and we wouldn't be wasting disk space like @mertalev mentioned (https://github.com/immich-app/immich/pull/6455#discussion_r1717809698). Would it be ok for me to implement?

Reply:
Yeah, I like that.

Reply:
Implemented by 1434979f9a85ba677164d34ffced13a4a258a7fb